realm/realm-jni/src/io_realm_internal_table.cpp

```diff
@@ -1710,9 +1728,11 @@ JNIEXPORT void JNICALL Java_io_realm_internal_Table_nativeMigratePrimaryKeyTable
     size_t number_of_rows = pk_table->size();
     for (size_t row_ndx = 0; row_ndx < number_of_rows; row_ndx++) {
         StringData table_name = pk_table->get_string(CLASS_COLUMN_INDEX, row_ndx);
-        if (table_name.begins_with(TABLE_PREFIX)) {
+        if (table_name.begins_with(table_prefix)) {
             // New string copy is needed, since the original memory will be changed.
-            pk_table->set_string(CLASS_COLUMN_INDEX, row_ndx, std::string(table_name.substr(strlen(TABLE_PREFIX))));
+            std::string str(table_name.substr(strlen(TABLE_PREFIX)));
```

Comment:
As we have already copied this into `table_prefix` can use `table_prefix.length()` instead of recalculating `strlen`